security/src/main/java/org/kaazing/gateway/security/auth/FileLoginModule.java

```diff
@@ -234,19 +226,19 @@ private void cleanState() {
         password = null;
     }
 
-    private boolean attemptAuthenticate(boolean useSharedState)
+    private void attemptAuthenticate(boolean useSharedState)
         throws LoginException {
 
         getUsernamePassword(useSharedState);
 
         if (username == null) {
-            LOG.debug("[FileLoginModule] " + "username not found, returning false");
-            return false;
+            LOG.debug("[FileLoginModule] username not found");
+            throw new LoginException("Username not found");
```

Comment:
I tried to answer below, throwing a LoginException from a login doesn't guarantee that it will show up in the  Http Security Filter. I can create the new LoginException and log that before throwing it, so that I don't duplicate the message. What do you think? I saw some login module implementations have a debug option to enable detailed logging, but I thought that using the debug level on the logger would simulate that.